curp/src/server/raw_curp/mod.rs

```diff
@@ -791,9 +775,104 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
+                .value(),
         )
     }
+
+    /// Apply conf changes and return true if self node is removed
+    #[allow(unused)] // TODO: remove
+    pub(super) fn apply_conf_change(
+        &self,
+        changes: Vec<ConfChange>,
+    ) -> Result<bool, ApplyConfChangeError> {
+        assert_eq!(changes.len(), 1, "Joint consensus is not supported yet");
+        let Some(conf_change) = changes.into_iter().next() else {
+            unreachable!("conf change is empty");
+        };
+
+        self.check_new_config(&conf_change)?;
+
+        Ok(self.switch_config(conf_change))
+    }
+
+    /// Check if the new config is valid
+    #[allow(clippy::unimplemented)] // TODO: remove
+    fn check_new_config(&self, conf_change: &ConfChange) -> Result<(), ApplyConfChangeError> {
+        let mut statuses_ids = self
+            .lst
+            .get_all_statuses()
+            .keys()
+            .copied()
+            .chain([self.id()])
+            .collect::<HashSet<_>>();
+        let mut config = self.cst.map_lock(|cst_l| cst_l.config.clone());
+        let conf_change_type =
+            ConfChangeType::from_i32(conf_change.change_type).unwrap_or_else(|| {
+                unreachable!("conf change type {} should valid", conf_change.change_type)
+            });
+        let node_id = conf_change.node_id;
+        match conf_change_type {
+            ConfChangeType::Add => {
+                if !statuses_ids.insert(node_id) || !config.insert(node_id) {
+                    return Err(ApplyConfChangeError::NodeAlreadyExists(node_id));
+                }
+            }
+            ConfChangeType::Remove => {
+                if !statuses_ids.remove(&node_id) || !config.remove(node_id) {
+                    return Err(ApplyConfChangeError::NodeNotExists(node_id));
+                }
+            }
+            ConfChangeType::Update => {
+                if statuses_ids.get(&node_id).is_none() || !config.contains(node_id) {
+                    return Err(ApplyConfChangeError::NodeNotExists(node_id));
+                }
+            }
+            ConfChangeType::AddLearner => {
+                unimplemented!("learner node is not supported yet");
+            }
+        }
+        if statuses_ids.len() < 3 || config.voters() != &statuses_ids {
+            return Err(ApplyConfChangeError::InvalidConfig);
+        }
+        Ok(())
+    }
+
+    /// Switch to a new config and return true if self node is removed
+    #[allow(clippy::unimplemented)] // TODO: remove
+    fn switch_config(&self, conf_change: ConfChange) -> bool {
+        let node_id = conf_change.node_id;
+        let conf_change_type =
+            ConfChangeType::from_i32(conf_change.change_type).unwrap_or_else(|| {
+                unreachable!("conf change type {} should valid", conf_change.change_type)
+            });
+        match conf_change_type {
+            ConfChangeType::Add => {
+                let member = Member::new(node_id, "", conf_change.address);
```

Comment:
The member name is empty here, should the ConfChange contains a member name?

Reply:
The `MemberAddRequest` of `etcd` does not contain `name`. add `name` here will break compatibility.